src/Main.cc

```diff
@@ -688,67 +621,71 @@ int main(int argc, const char* argv[]) {
             }
         }
 
-        host = host.empty() ? "0.0.0.0" : host;
-        port = (port < 0) ? DEFAULT_SOCKET_PORT : port;
         bool port_ok(false);
 
-        if (port == DEFAULT_SOCKET_PORT) {
+        if (settings.port < 0) {
+            settings.port = DEFAULT_SOCKET_PORT;
             int num_listen_retries(0);
             while (!port_ok) {
                 if (num_listen_retries > MAX_SOCKET_PORT_TRIALS) {
-                    spdlog::error("Unable to listen on the port range {}-{}!", DEFAULT_SOCKET_PORT, port - 1);
+                    spdlog::error("Unable to listen on the port range {}-{}!", DEFAULT_SOCKET_PORT, settings.port - 1);
                     break;
                 }
-                app.listen(host, port, LIBUS_LISTEN_EXCLUSIVE_PORT, [&](auto* token) {
+                app.listen(settings.host, settings.port, LIBUS_LISTEN_EXCLUSIVE_PORT, [&](auto* token) {
                     if (token) {
                         port_ok = true;
                     } else {
-                        spdlog::warn("Port {} is already in use. Trying next port.", port);
-                        ++port;
+                        spdlog::warn("Port {} is already in use. Trying next port.", settings.port);
+                        ++settings.port;
```

Comment:
Just personal taste: I'd instead use `auto port = settings.port`, then update `port` if it is already in use. I like to keep the initial settings unchanged. Of course, you don't need to follow this comment :).

Reply:
Yes, that's probably a good idea

Reply:
Have just looked at the code again... introducing a `port` variable might be more trouble than it's worth, because `settings.port` is used elsewhere. Changing that would require more refactoring, and I think this PR has already ballooned a bit too much.